components/gitpod-protocol/src/accounting-protocol.ts

```diff
@@ -199,6 +199,9 @@ export namespace Subscription {
     export function isActive(s: Subscription, date: string): boolean {
         return s.startDate <= date && (s.endDate === undefined || date < s.endDate);
     }
+    export function isCancelled(s: Subscription, date: string): boolean {
+        return s.cancellationDate !== undefined && s.cancellationDate < date;
```

Comment:
Nit: It would seem safer to check for anything "falsy" (for example, empty string `""` or `null`) rather than just specifically `undefined` here:

```suggestion
        return !!s.cancellationDate && s.cancellationDate < date;
```

Reply:
Also, can there be any cases where you don't have a `cancellationDate` but you do have an `endDate`? 👀

Reply:
Yeah, both valid points. It _should_ be undefined, and it _should_ always have a `cancellationDate`. But that's one of the tables that saw a lot of manual "fixes", so worth being more conservative here! :100:

Reply:
Checked the prod DB: we actually have 6 cases where this is not set :see_no_evil: Made the code fail-safe to save us our :brain:s.